### PR TITLE
PivotGridDataSource: Describe how to apply date formats to field values

### PR DESCRIPTION
**Pull request by RomanTsukanov**
Fixes the following docissues:

https://supportcenter.devexpress.com/internal/ticket/details/T905614
https://supportcenter.devexpress.com/internal/ticket/details/T855330
https://supportcenter.devexpress.com/internal/ticket/details/T901139
https://supportcenter.devexpress.com/internal/ticket/details/T893707
https://supportcenter.devexpress.com/internal/ticket/details/T893709

### CODE REVIEW

**Review by LexDX — APPROVED** [review]

**Review by AlisherAmonulloev — APPROVED** [review]

File: api-reference/30 Data Layer/PivotGridDataSource/1 Configuration/fields/format.md [hunk 1]
```diff
@@ -162,6 +162,207 @@ See the [format](/api-reference/50%20Common/Object%20Structures/format '/Documen
 
 ---
 
+Date formats require additional configuration. The **PivotGridDataSource** groups date field values by date component: Year-Quarter-Month. For this, the values are converted from dates to numbers, but numbers cannot be formatted using date formats. To apply a date format in this case, you need to override this grouping.
```
Comment (AlexJameson) [inline api-reference/30 Data Layer/PivotGridDataSource/1 Configuration/fields/format.md]: I might be out of context, but what is 'date component'? Is it a legit term?

Can we also replace 'numbers cannot be formatted using date formats' with smth like 'you cannot apply date formats to numeric values'?

File: api-reference/30 Data Layer/PivotGridDataSource/1 Configuration/fields/format.md [hunk 2]
```diff
@@ -162,6 +162,207 @@ See the [format](/api-reference/50%20Common/Object%20Structures/format '/Documen
...
+
+The following code declares a custom group for the `ShippingDate` data field. This configuration overrides this field's default grouping. Field values are grouped first by year, then by month. Months are formatted using the `MMM` format (`Jan`, `Feb`, `Mar`, etc.). This format is applied to the Date values returned from the [selector](/Documentation/ApiReference/Data_Layer/PivotGridDataSource/Configuration/fields/#selector) function.
```
Comment (AlexJameson) [inline api-reference/30 Data Layer/PivotGridDataSource/1 Configuration/fields/format.md]: A possible option: Months are formatted using the `MMM` format -> Month are formatted with the `MMM` format.

Example from the Microsoft docs: [By default, all worksheet cells are formatted with the General number format](https://docs.microsoft.com/en-us/office/troubleshoot/excel/format-cells-settings).

File: api-reference/30 Data Layer/PivotGridDataSource/1 Configuration/fields/format.md (hunk 1 above)
Comment (RomanTsukanov) [reply api-reference/30 Data Layer/PivotGridDataSource/1 Configuration/fields/format.md]: > what is 'date component'?

Year, quarter, month.

> Can we also replace 'numbers cannot be formatted using date formats' with smth like 'you cannot apply date formats to numeric values'?

This kinda breaks the flow in the sentence. I would prefer to keep it as is.

**Review by AlexJameson — APPROVED** [review]

File: api-reference/30 Data Layer/PivotGridDataSource/1 Configuration/fields/format.md (hunk 1 above)
Comment (dirk-pieterse) [inline api-reference/30 Data Layer/PivotGridDataSource/1 Configuration/fields/format.md]:
```suggestion
Date formats require additional configuration. The **PivotGridDataSource** groups date field values by date component: Year-Quarter-Month. For this, the values are converted from dates to numbers, but you cannot use date formats to format numbers. To apply a date format in this case, override this grouping.
```

File: api-reference/30 Data Layer/PivotGridDataSource/1 Configuration/fields/format.md [hunk 3]
```diff
@@ -162,6 +162,207 @@ See the [format](/api-reference/50%20Common/Object%20Structures/format '/Documen
 
 ---
 
+Date formats require additional configuration. The **PivotGridDataSource** groups date field values by date component: Year-Quarter-Month. For this, the values are converted from dates to numbers, but numbers cannot be formatted using date formats. To apply a date format in this case, you need to override this grouping.
+
+The following code declares a custom group for the `ShippingDate` data field. This configuration overrides this field's default grouping. Field values are grouped first by year, then by month. Months are formatted with the `MMM` format (`Jan`, `Feb`, `Mar`, etc.). This format is applied to the Date values returned from the [selector](/Documentation/ApiReference/Data_Layer/PivotGridDataSource/Configuration/fields/#selector) function.
```
Comment (dirk-pieterse) [inline api-reference/30 Data Layer/PivotGridDataSource/1 Configuration/fields/format.md]:
```suggestion
The following code declares a custom group for the `ShippingDate` data field. This configuration overrides this field's default grouping. Field values are grouped by year and then by month. Months are formatted as `MMM` (`Jan`, `Feb`, `Mar`, etc.). This format is applied to the Date values the [selector](/Documentation/ApiReference/Data_Layer/PivotGridDataSource/Configuration/fields/#selector) function returns.
```

**Review by dirk-pieterse — APPROVED** [review]

**Pull request merged**